File: Teledyne DALSA/Sapera/Demos/NET/GigEFlatFieldDemo/VB/SapMyProcessing.vb
Class SapMyProcessing
   Inherits SapProcessing
   Private m_pFlatField As SapFlatField
   ' Constructor

   Public Sub New(ByVal pBuffers As SapBuffer, ByVal pFlatField As SapFlatField, ByVal pCallback As SapProcessingDoneHandler, ByVal pContext As Object)
      MyBase.New(pBuffers)
      MyBase.ProcessingDoneEnable = True
      AddHandler MyBase.ProcessingDone, pCallback
      MyBase.ProcessingDoneContext = pContext
      m_pFlatField = pFlatField
   End Sub



   Public Overloads Overrides Function Run() As Boolean
      ' Check if flat field correction is enabled and if it's to be done by software
      If m_pFlatField.Enabled AndAlso m_pFlatField.SoftwareCorrection Then
         ' Do software flat field correction
         m_pFlatField.Execute(MyBase.Buffer, MyBase.Index)
      End If
      Return True
   End Function
End Class

File: Teledyne DALSA/Sapera/Demos/NET/FlatFieldDemo/CSharp/SapMyProcessing.cs
using System;
using System.Collections.Generic;
using System.Text;
using DALSA.SaperaLT.SapClassBasic;

namespace DALSA.SaperaLT.Demos.NET.CSharp.FlatFieldDemo
{
   class SapMyProcessing :  SapProcessing
   {
      private SapFlatField m_pFlatField;
     // Constructor

     public SapMyProcessing(SapBuffer pBuffers, SapFlatField pFlatField, SapProcessingDoneHandler  pCallback, Object pContext)
        : base(pBuffers)
     {
        base.ProcessingDoneEnable = true;
        base.ProcessingDone += pCallback;
        base.ProcessingDoneContext = pContext;
        m_pFlatField = pFlatField;
     }

 

     public override bool Run()
     {
         // Check if flat field correction is enabled and if it's to be done by software
         if (m_pFlatField.Enabled && m_pFlatField.SoftwareCorrection)
         {
            // Do software flat field correction
            m_pFlatField.Execute(base.Buffer, base.Index);
         }
         return true;
     }

   
   }
}
